### ISSUE TITLE
GitFinder does not handle includes in the .gitconfig very well

### ISSUE THREAD

**Issue by unikitty37** [milke/GitFinder-Issues#80]
My `.gitconfig` file includes the following:

```
[includeIf "gitdir/i:~/"]
        path = ~/.gitconfig-personal
[includeIf "gitdir/i:~/Development/_Public/"]
        path = ~/.gitconfig-github
```

When I open a repo in my home folder, I see the following:

![Screenshot 2021-07-27 at 09 28 51](https://user-images.githubusercontent.com/139183/127122646-62889d18-be61-4c97-89a0-8e2ca3c5bd04.png)

When I click `Open "includes" configuration…` I see the following:

![image](https://user-images.githubusercontent.com/139183/127122714-f3e54100-536a-4172-a546-11eb0d707a81.png)

The file is not listed, and there is no Add button.

As a result, the GitFinder integration does not seem to work *at all* — I see no badges in the Finder. Opening a repo outside of my home folder does not produce these dialogs, and the integration works.

I'm also not entirely 100% sure on what I *should* be doing — the phrase `Select in "includes" configuration included file at path` doesn't really read very well.

v 1.7.1 (104)
macOS 10.14.6 (18G9216)

**Comment by milke** [comment]
Yes, you're right. This can be considered a bug. As you can see, GitFinder correctly recognises that configuration file contains `include` directives. But when implementing it, I completely overlooked conditional `includeIf` case 🙁 That's the reason you see empty include directive list. If it worked correctly, you would see something like this (in your particular case):
<img width="515" alt="Screen Shot 2021-07-27 at 13 44 52" src="https://user-images.githubusercontent.com/291305/127149587-9643861f-b0b6-4b77-8cf4-27780e599837.png">
That would make it clear what exactly to do. GitFinder guides you with help buttons, for example like this:
<img width="612" alt="Screen Shot 2021-07-27 at 13 45 06" src="https://user-images.githubusercontent.com/291305/127149794-4ebd4886-de1a-42e7-b965-4685e857cb90.png">
GitFinder would even "force' you to select the correct file. If you try to select anything else, you will get something like this:
<img width="912" alt="Screen Shot 2021-07-27 at 13 45 25" src="https://user-images.githubusercontent.com/291305/127149904-1f29ed73-5180-46c2-accb-2fc893af3e26.png">
Once you do everything right, you'd see something like this:
<img width="515" alt="Screen Shot 2021-07-27 at 13 45 42" src="https://user-images.githubusercontent.com/291305/127150098-7ad137bb-e3ee-4c3b-87b6-24df41705265.png"> 
Finally, GitFinder is capable of handling nested include directives (included file containing further include directives) and the nest lever is indicated in the UI. For example, like this:
<img width="515" alt="Screen Shot 2021-07-27 at 13 47 51" src="https://user-images.githubusercontent.com/291305/127150273-ddfd5324-3210-4795-90b0-a047236de7d0.png">
Of course, all this if it worked 😂. I'll try to make a beta version handling conditional includes and let you know when it's available, so you can test it out.

And like I've noted in some other threads, this all should usually work out of the box, but since GitFinder is fully sandboxed, you need to explicitly tell it where the included files are, so that they can be included into GitFinder's sandbox. Fortunately, it's just one time operation.

**Comment by milke** [comment]
Okay, I've just uploaded new beta version **1.7.2b1 (build 105)**. It should handle conditional includes in configuration files correctly. Please try it out and let me know if it does (or doesn't) work fine. You can download the new beta [**here**](https://get.gitfinder.com/GitFinder_beta.dmg), but you can also upgrade your current copy automatically, just select **Beta** release channel, like below:
<img width="495" alt="Screen Shot 2021-07-27 at 17 20 57" src="https://user-images.githubusercontent.com/291305/127181875-6b6e2ef7-05c1-4c33-94a1-f34e716cafae.png">

**Comment by unikitty37** [comment]
That's working as expected — thanks!

**Comment by milke** [comment]
The fix is now officially available in version **1.7.2 (build 108)**, so I'm closing this.
